chia/wallet/wallet_state_manager.py

```diff
@@ -1398,6 +1401,23 @@ async def new_coin_state(
                     await self.retry_store.remove_state(coin_state)
                 continue
 
+    async def new_coin_state(
+        self,
+        coin_states: List[CoinState],
+        peer: WSChiaConnection,
+        fork_height: Optional[uint32],
+    ) -> bool:
+        try:
```

Comment:
Maybe later we can move the rollback recovery out here, as well.

Reply:
This would require splitting it up into also a function which handles single coin states because the rollback recovers is done per loop cycle. But i will come up with some other cleanups for this method anyway and there are already open PRs for it!